### ISSUE TITLE
Problem with BooleanEquals

### ISSUE THREAD

**Issue by mackensen** [udondan/cdk-ssm-document#15]
Similar issue to #14, this is valid JSON for a choice step assuming that `DoSomething` was defined as Boolean input parameter:

```json
             {
                "name": "DoSomethingCheck",
                "action": "aws:branch",
                "inputs": {
                   "Choices": [
                      {
                         "NextStep": "DoSomething",
                         "Variable": "{{DoSomething}}",
                         "BooleanEquals": true
                      },
                      {
                         "NextStep": "DoSomethingElse",
                         "Variable": "{{DoSomething}}",
                         "BooleanEquals": false
                      }
                   ]
                }
             },
```

This fails for the same reason: "Failed to update resource. Input true is of type String, but expected type is Boolean."

**Comment by udondan** [comment]
Hi Charles,

does this actually work when you define the doc in the console? After I fixed #14 I was able to create documents with boolean parameters, but was unable to execute them. When executing, it complained that Boolean is an invalid parameter type... I contacted the AWS support and their response was quite interesting:

> We stated it in our documentation that you can use Boolean type but have not seen it work

So as I see it, fixing this will only allow you to create a document which cannot be executed.

AWS support suggested to only use String parameters and limit the possible values to "true" and "false".

Daniel

**Comment by udondan** [comment]
Ah wait, `aws:branch`... that's an Automation document. I haven't tried that. Above experience is related to Command documents. Maybe boolean types are supported in Automation documents.

**Comment by udondan** [comment]
The fix has been released with v2.1.2

**Comment by mackensen** [comment]
Yes, I'm sorry, this was in an Automation document, and it did work from the console. Thanks for the quick response, I've verified that it's working now.
